### PR TITLE
fix(featuremedia): Fix drag and drop to feature media an archived con…

### PR DESCRIPTION
**Pull request by ioanpocol**
…tent item (search -> archived)

### CODE REVIEW

File: scripts/apps/authoring/authoring/directives/ItemAssociationDirective.js [hunk 1]
```diff
@@ -47,7 +47,10 @@ export function ItemAssociationDirective(superdesk, renditions, config, authorin
             function getItem(event, dataType) {
                 let item = angular.fromJson(event.originalEvent.dataTransfer.getData(dataType));
 
-                if (item._type !== 'externalsource') {
+                if (item._type === 'archived') {
+                    return api.find('archived', item._id)
```
Comment (petrjasek) [inline scripts/apps/authoring/authoring/directives/ItemAssociationDirective.js]: seems like this `.then()` is useless there.

File: scripts/apps/authoring/authoring/directives/ItemAssociationDirective.js [hunk 2]
```diff
@@ -47,7 +47,10 @@ export function ItemAssociationDirective(superdesk, renditions, config, authorin
...
+                        .then((result) => result);
+                } else if (item._type !== 'externalsource') {
                     return api.find('archive', item._id)
```
Comment (petrjasek) [inline scripts/apps/authoring/authoring/directives/ItemAssociationDirective.js]: maybe it should just do `if (item._type !== 'externalsource') { return api.find(item._type, item._id); }`

**Comment by jerome-poisson** [comment]
@ioanpocol is this patch fixing the issue for context panel too? cf. https://dev.sourcefabric.org/secure/attachment/32138/archived_picture_bug.webm at the end

**Comment by ioanpocol** [comment]
Now yes

**Pull request merged**